apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/nodes/NodesWebscript1.java

```diff
@@ -651,6 +657,96 @@ public void setParent(@UriVariable final String space, @UriVariable final String
 
     }
 
+    @ApiOperation(value = "Retrieves all comments for a given node")
+    @Uri(value = "/nodes/{space}/{store}/{guid}/comments", method = HttpMethod.GET)
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 403, message = "Not Authorized"),
+            @ApiResponse(code = 404, message = "Not Found")
+    })
+    public void getComments(@UriVariable String space, @UriVariable String store, @UriVariable String guid,
+            @RequestParam int skipcount, @RequestParam int pagesize, WebScriptResponse response) throws IOException {
+        final NodeRef target = new NodeRef(space, store, guid);
+        if (nodeService.exists(target)) {
+            if(permissionService.hasPermission(target, PermissionService.READ)) {
+                Conversation comments = commentService.getComments(target, skipcount, pagesize);
+                boolean canCreate = permissionService.hasPermission(target, PermissionService.CREATE_CHILDREN);
+                comments.setCanCreate(canCreate);
+                writeJsonResponse(response, comments);
+            } else {
+                throw new AccessDeniedException("User does not have permission to read parent node");
+            }
+        } else {
+            writeNotFoundResponse(response, target);
+        }
+    }
+
+    @ApiOperation(value = "Appends a new comment to the given node.")
+    @Uri(value = "/nodes/{space}/{store}/{guid}/comments", method = HttpMethod.POST)
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 403, message = "Not Authorized"),
+            @ApiResponse(code = 404, message = "Not Found")
+    })
+    public void postComment(@UriVariable String space, @UriVariable String store, @UriVariable String guid,
+            WebScriptRequest request, WebScriptResponse response) throws IOException {
+        final NodeRef target = new NodeRef(space, store, guid);
+        if (nodeService.exists(target)) {
+            ObjectMapper objectMapper = new ObjectMapper();
+            String content = objectMapper.readTree(request.getContent().getInputStream()).textValue();
```

Comment:
I'm a bit confused about the purpose of this line. We're reading in the content of the request as a JsonNode, just to get the text value of it. Isn't that equivalent to stripping off the leading and trailing quotation marks of the content?

What json structure is the payload of this endpoint supposed to conform to anyway? Is there a swagger spec for this?

Reply:
there is a spec on the ticket. Im not entirely sure where to put it in the repo though

Reply:
I dont get the remark over stripping though?

Reply:
> Isn't that equivalent to stripping off the leading and trailing quotation marks of the content?

No, because if you have `"some text\"some more text"`, this must decode to `some text"some more text`, and not to `some text\"some more text`.

Reply:
Also, this is not according to spec. 
The request  body is JSON and should have a `content` property (a very trimmed down version of the  `Comment` class that only has the `content` field.)

Reply:
After reading the specand what Lars said it makes more sense. 
It's probably not worth another POJO but it should be something like 
```
JsonNode jsonNode = objectMapper.readTree(request.getContent().getInputStream());
if (jsonNode.has("content")) {
    String content = jsonNode.get("content").textValue();
    // ... continue
} else {
    // send an error about the missing field
}
```

Reply:
Not using a POJO will break the automatic generation of a proper swagger spec from apix. (Or are we no longer doing that?)

Reply:
I'm not sure, but it's probably good to keep that option open, so using a POJO instead of what I wrote.